rotkehlchen/exchanges/exchange.py

```diff
@@ -214,32 +214,23 @@ def query_online_deposits_withdrawals(
         """Queries the exchange's API for the asset movements of the user
 
         Should be implemented in subclasses.
+
+        Deprecated, will be replaced by query_online_history_events
         """
-        raise NotImplementedError(
-            'query_online_deposits_withdrawals should only be implemented by subclasses',
-        )
+        return []
 
-    def query_online_income_loss_expense(
+    def query_online_history_events(
             self,
             start_ts: Timestamp,
             end_ts: Timestamp,
-    ) -> list['HistoryEvent']:
-        """Queries the exchange's API for simple history events of the user
+    ) -> Sequence['HistoryBaseEntry']:
+        """Queries the exchange's API for history events of the user
 
-        Should be implemented in subclasses.
-        Has to be implemented by exchanges if they have anything exchange specific
-
-        For example coinbase
+        Should be implemented in subclasses, unless query_history_events is reimplemented with
```

Comment:
what do you mean by this?

Reply:
`query_online_history_events` normally needs implemented in subclasses since its called by `query_history_events`. The exception to this is when the subclass reimplements `query_history_events` with custom logic (for instance in kraken) that may not necessarily use the `query_online_history_events` function.